src/main/java/io/fabric8/maven/Maven.java

```diff
@@ -44,17 +45,31 @@ public static Model readModel(String path) {
      * @return the maven {@link Model}
      */
     public static Model readModel(Path pom) {
-        Model model;
-        MavenXpp3Reader reader = new MavenXpp3Reader();
         try (BufferedReader br = Files.newBufferedReader(pom)) {
-            model = reader.read(br);
+            Model model = readModel(br);
             model.setPomFile(pom.toFile());
+            return model;
+        } catch (IOException io) {
+            throw new UncheckedIOException("Error while reading pom.xml", io);
+        }
+    }
+
+    /**
+     * Read the {@link Path} as a {@link Model}
+     *
+     * @param rdr a Reader on the contents of a pom file
+     * @return the maven {@link Model}
+     */
+    public static Model readModel(Reader rdr) {
```

Comment:
In this case we better add a null check in `writeModel(Model)`, since the model.getPomFile() may not be set